e2e/pages/widgets/shared/input.js

```diff
@@ -0,0 +1,39 @@
+export class Input {
+  static byTestId = testId => {
+    return new Input(() => cy.findByTestId(testId));
+  };
+
+  static byLabel = label => {
+    return new Input(() => cy.findByLabelText(label));
+  };
+
+  constructor(locator) {
+    this.locator = locator;
+  }
+
+  setValue = value => {
+    this.clear();
+
+    if (value !== "") {
+      this.append(value);
+    }
+
+    return this;
+  };
+
+  append = value => {
+    this.locator().type(value);
+    return this;
+  };
+
+  clear = value => {
```

Comment:
Does clear accept `value`? I think not, in that case we should remove this probably